pkg/api/meta/meta.go

```diff
@@ -337,6 +443,7 @@ type genericAccessor struct {
 	deletionTimestamp **unversioned.Time
 	labels            *map[string]string
 	annotations       *map[string]string
+	ownerReferences   sliceAccessor
```

Comment:
Now that I see that there are multiple `OwnerReference` types, this would probably have to still be a `reflect.Value`. `sliceAccessor` could go away, but the setter/getter below would still be just as complicated.